lib/util/src/err_ctx.c

```diff
@@ -0,0 +1,19 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2022 Micron Technology, Inc.  All rights reserved.
+ */
+
+#include <stddef.h>
+
+#include <hse/types.h>
+
+const char *
+err_ctx_strerror(const enum hse_err_ctx ctx)
+{
+    switch (ctx) {
+    case HSE_ERR_CTX_NONE:
+        return NULL;
+    }
+
+    return NULL;
+}
```

Comment:
Not sure why we need an enum, just expose that hse_err_ctx is an int16_t ???
All that really matters is the conversion function, I would say if I have a stringify function then I don't really care what the context is...  Just make it an int.  Some projects will have longs, others like ours have uint16_t...

Reply:
I'll take an int and cast in the switch declaration that way we can avoid casts. Will that suit you?

Reply:
You would have to modify the source to generate a string anyways. I am confused. If it is an enum, then gcc will warn you if you didn't provide a case for a variant.

Reply:
I was thinking caller would provide a function outside of hse so they don't have to modify hse, no one wants to have to do that...

Reply:
Yes, that is the case when your context is a CURLCode, but we have `enum hse_err_ctx` in `hse/types.h`

Reply:
An enum makes sense for `libhse`.  This is where `libhse` will define codes to indicate "transaction timed out", or "transaction write conflict".  Other users (like the app that uses it for CURLCodes) can use a different approach.

Reply:
Would be easier to use this API if it returned `""` or `"no error"` instead of NULL.

Reply:
I really don't understand the point of defining it.  It should just be an opaque arg that is passed to the "stringify" function, HSE should never look at it otherwise...  I.e., if you have a stringify function ptr then call it with the ctx arg, otherwise just completely ignore the ctx arg.  The enum is just more useless clutter as far as I can tell, I've asked six times what the purpose of the enum is and have not gotten any answer that explains why it's needed....